diff --git a/src/Makefile.am b/src/Makefile.am
index a27f961..ca4e638 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -21,9 +21,9 @@ AM_LDFLAGS =\

 AM_LDADD = $(AC_LDADD)

-bin_PROGRAMS = idevicerestore
+lib_LTLIBRARIES = libidevicerestore.la

-idevicerestore_SOURCES = idevicerestore.c common.c tss.c fls.c mbn.c img3.c img4.c ftab.c ipsw.c normal.c dfu.c recovery.c restore.c asr.c fdr.c limera1n.c download.c locking.c socket.c thread.c jsmn.c json_plist.c
-idevicerestore_CFLAGS = $(AM_CFLAGS)
-idevicerestore_LDFLAGS = $(AM_LDFLAGS)
-idevicerestore_LDADD = $(AM_LDADD)
+libidevicerestore_la_SOURCES = libidevicerestore.c common.c tss.c fls.c mbn.c img3.c img4.c ftab.c ipsw.c normal.c dfu.c recovery.c restore.c asr.c fdr.c limera1n.c download.c locking.c socket.c thread.c jsmn.c json_plist.c
+libidevicerestore_la_CFLAGS = $(AM_CFLAGS)
+libidevicerestore_la_LDFLAGS = $(AM_LDFLAGS)
+libidevicerestore_la_LIBADD = $(AM_LDADD)
diff --git a/src/idevicerestore.c b/src/libidevicerestore.c
similarity index 100%
rename from src/idevicerestore.c
rename to src/libidevicerestore.c
